forgot_password/handlers.py

```diff
@@ -22,10 +27,66 @@
 from .util import email as emailutil
 from .util import user as userutil
 
-
 logger = logging.getLogger(__name__)
 
 
+class IllegalArgumentError(ValueError):
+    pass
+
+
+ResetPasswordRequestParams = namedtuple(
+    'ResetPasswordRequestParams',
+    ['code', 'user_id', 'expire_at', 'user', 'user_record'])
+
+
+def download_template(url, name):
+    path_prefix = 'templates/forgot_password'
+    path = '/'.join([path_prefix, name])
+
+    Path(path_prefix).mkdir(parents=True, exist_ok=True)
```

Comment:
Should I add one more line in `jinja_env()` in `forgot_password/template.py` so that the template resolve order will be:
Downloaded template > User provided template > Default template.